docs/map.rst

```diff
@@ -0,0 +1,602 @@
+
+
+Removing an entry from a map
+""""""""""""""""""""""""""""
+
+::
+
+    Map.delete :: Ord k => k -> Map k v -> Map k v
+    Map.delete key m = ...
+
+:haddock_short:`Data.Map.Strict#delete` removes the entry with the specified
+``key`` from the map ``m``.  If the key doesn't exist it leaves the map
+unchanged. Remember, maps are immutable so if you delete an entry from a map you
+need to assign the new map to a new variable.
+
+::
+
+    Map.delete 1 Map.empty
+    > Map.empty
+
+    Map.delete 1 (Map.fromList [(1,"one"),(2,"two"),(3,"three")])
+    > fromList [(2,"two"),(3,"three")]
+
+Filtering map entries
+"""""""""""""""""""""
+
+::
+
+    Map.filterWithKey :: (k -> v -> Bool) -> Map k v -> Map k v
+    Map.filterWithKey predicate m = ...
+
+:haddock_short:`Data.Map.Strict#filterWithKey` produces a map consisting of all
+entries of ``m`` for which the ``predicate`` returns ``True``.
+
+::
+
+    let f key value = key == 2 || value == "one"
+    Map.filterWithKey f (Map.fromList [(1,"one"), (2,"two"), (3,"three")])
+    > fromList [(1,"one"),(2,"two"]
+
+
+Modifying a map entry
+"""""""""""""""""""""
+
+::
+
+    Map.adjust :: Ord k => (v -> v) -> k -> Map k v -> Map k v
+    Map.adjust f key m = ...
+
+:haddock_short:`Data.Map.Strict#abjust` applies the value transformation
+function ``f`` to the entry with given ``key``. If no entry for that key exists
+then the map is left unchanged.
+
+::
+
+    Map.alter :: Ord k => (Maybe v -> Maybe v) -> k -> Map k v -> Map k v
+    Map.alter f key m = ...
+
+Apply the value transformation function ``f`` to the entry with given ``key``,
+if no entry for that key exists then the function is passed ``Nothing``. If the
+function returns ``Nothing`` then the entry is deleted, if the function returns
+``Just v2`` then the value for the ``key`` is updated to ``v2``. In other words,
+alter can be used to insert, update, or delete a value.
+
+::
+
+    let removeElement _ = Nothing
+    Map.alter removeElement "key" (Map.fromList [("key", 0)])
+    > fromList []
+
+    let setValueToOne _ = Just 1
+    Map.alter setValueToOne "key" (Map.fromList [("key", 0)])
+    > fromList [("key",1)]
+
+    Map.alter setValueToOne "key" Map.empty
+    > fromList [("key",1)]
+
+Modifying all map entries (mapping)
+"""""""""""""""""""""""""""""""""""
+
+::
+
+    Map.map :: (a -> b) -> Map k a -> Map k v
+    Map.map f m = ...
+
+:haddock_short:`Data.Map.Strict#map` creates a new map by applying the
+transformation function ``f`` to each entries value. This is how `Functor
+<https://wiki.haskell.org/Typeclassopedia#Functor>`_ is defined for maps.
+
+::
+
+    Map.map (*10) (Map.fromList [("haskell", 45), ("idris", 15)])
+    > fromList [("haskell",450),("idris",150)]
+
+    -- Use the Functor instance for Map.
+    (*10) <$> Map.fromList [("haskell", 45), ("idris", 15)]
+    > fromList [("haskell",450),("idris",150)]
+
+There are several other more complex mapping functions available that let you
```

Comment:
I think both `mapWithKey` and `traverseWithKey` probably deserve to be in this tutorial. For `traverseWithKey`, you could give a simple example traversing in `IO`:

```haskell
-- | Ask the user how they want to schedule a bunch of tasks
-- that the boss has assigned certain priorities.
makeSchedule :: Map Task Priority -> IO (Map Task DateTime)
makeSchedule = traverseWithKey $ \task priority ->
  do
    putStrLn $ "The boss thinks " ++ show task ++ " has priority " ++ show priority ++
                      ". When do you want to do it?"
    readLn
```

Reply:
Good idea!